consensus/src/common.rs

```diff
@@ -26,6 +31,12 @@ where
     fn length(&self) -> usize;
     fn last_vrf(&self) -> Option<String>;
     fn state_hash(&self) -> Option<StateHash>;
+    fn gen_spot(block: &mut ProtocolState);
```

Comment:
Why have functions without self in this trait? It's not OO design and I don't really understand why they fit in the trait

Reply:
previous review feedback mentioned why not integrate gen_spot in a trait or struct and you said it's more about OO design. In my previous merge, you speak about integrating these methods into Chain traits for better OO design. Then I just think maybe it's a good idea to put them into the Chain trait. In the case of is_short_range() there are two arguments how can they fit in self in Rust? It will be good if you can let me know what you prefer for a better OO design. A reference will be very nice.

Reply:
it's not good idea to use OO design here